### PR TITLE
Feat/hng 46 faq accordion component

### LINKED ISSUE #46: [FEATURE] Create Accordion Component
## Description
Develop an Accordion Component. This component will be implemented in the pricing page to list all FAQs. The Accordion component should not drop down more than 1 item.

## Acceptance Criteria

- The accordion component should be implemented according to the design [Use this as guideline](https://ui.shadcn.com/docs/components/accordion)
- Ensure the accordion component is responsive in all screen sizes
- Write test for your component

### Requirements
**The prop should receive the Accordion Header, the question, and the Accordion content, the response**

### Test Script

- Ensure the component renders the requirements accordingly
- Ensure the component receives the accordion header, the question.
- Ensure the component receives the accordion content, the response.

## Tasks

- [x] Develop the component based on the requirements and design
- [x] Make component responsive on all screens
- [x] Write unit test for the component

### Link
[Accordion](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=4-20796&m=dev)

### PR DESCRIPTION
**Pull request by frank1003A**
# Description

Closes #46

## What were you told to do?

Implement the FAQ Accordion component as described in [issue #46](https://github.com/hngprojects/hng_boilerplate_nextjs/issues/46).

## What did you do?

- Added the `FaqAccordion` component.
- Updated styles and classes for the accordion component.
- Added unit tests to verify accordion functionality.
- Ensured responsiveness of the accordion component.
- Updated the TypeScript interfaces and component props to support additional customizations.
- Fixed any issues related to UI interaction and state management.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x ] My code follows the code style of this project.
- [ x] This PR does not contain plagiarized content.
- [x ] The title and description of the PR are clear and explain the approach.
- [ x] I am making a pull request against the **dev branch** (left side).
- [x ] My commit message style matches our requested structure.
- [ x] My code additions do not fail code linting checks or unit tests.
- [x ] I am only making changes to files I was requested to.

# Screenshots/Videos

ts1
ts2
ts4
ts3

### CODE REVIEW

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable prettier/prettier */
```
Comment (taiwonaf) [inline src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: why did you disable this ?

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      "Yes, the application is accessible from multiple devices including desktops, tablets, and smartphones. Your data will be synced across all devices, ensuring a seamless experience.",
+  },
+];
+
+describe("faqAccordion", () => {
+  it("renders correctly with given FAQ data", () => {
+    expect.hasAssertions();
+    render(<FaqAccordion faqs={faqs} />);
+
+    // Check if the questions are rendered
```
Comment (taiwonaf) [inline src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: remove comment

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  it("renders correctly for different screen sizes", () => {
+    expect.hasAssertions();
+    render(
+      <FaqAccordion
+        faqs={faqs}
+        containerClassName="p-6 bg-neutral-50 w-full md:w-96 rounded-xl"
+      />,
+    );
+
+    // Check if the container class includes responsive width
```
Comment (taiwonaf) [inline src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: remove

**Review by taiwonaf — CHANGES_REQUESTED** [review]

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx (hunk 3 above)
Comment (frank1003A) [reply src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: done

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx (hunk 2 above)
Comment (frank1003A) [reply src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: done

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx (hunk 1 above)
Comment (frank1003A) [reply src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: Eslint configuration with local was giving issues. Fixed now

**Review by taiwonaf — APPROVED** [review]

**Pull request merged**